students/laud/session04/kata_fourteen.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import sys
+import random
+import string
+
+# Strip out punctuations, etc
+def cleanup(text):
+    lowercase = text.lower()
+    exclude = set(string.punctuation)
+    out = ''.join( ch for ch in lowercase if ch not in exclude)
+    return out
+
+# Read txt file
+def read_in_data(filename):
+    with open(filename, 'r') as text:
+        data = text.read()
+        clean = cleanup(data)
+    return clean
+
+# Split text into list
+def make_words(in_data):
+    data = in_data.split()
+    return data
+
+# Build key-value pairs
+def build_trigram(words):
+    trigrams = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        trigrams.setdefault(pair, []).append(follower)
+    return trigrams
+
+# Generate new text
+def build_text(word_pairs, limit):
+    word_list = []
+    starting_pair = list(random.choice(list(word_pairs)))
+    word_list.extend(starting_pair)
+    for x in range(int(limit)):
+        current_key = tuple(word_list[-2:])
+        if current_key in word_pairs:
+            next_word = random.choice(list(word_pairs[current_key]))
+            word_list.append(next_word)
+        else:
+            continue
```

Comment:
This is effectively the same as `break`, because continuing the loop will keep seeing the same last 2 words as the `current_key`, which won't be in `word_pairs` again (because nothing changed), so keep hitting this `else` block repeatedly until the `range(int(limit))` is exhausted. Better just use `break`.